torchprime/torch_xla_models/train.py

```diff
@@ -66,18 +63,17 @@ def __init__(
     self.train_dataset = train_dataset
 
     # Set up SPMD mesh and shard the model
-    num_devices = xr.global_runtime_device_count()
-    xs.set_global_mesh(
-      xs.Mesh(
-        np.array(range(num_devices)),
-        (num_devices, 1),
-        axis_names=("fsdp", "tensor"),
-      )
-    )
-    logger.info(f"Logical mesh shape: {xs.get_global_mesh().shape()}")
-    self.input_sharding_spec = xs.ShardingSpec(
-      xs.get_global_mesh(), ("fsdp", None), minibatch=True
+    dcn_mesh_shape = (config.mesh.dcn, 1, 1, 1)
+    ici_mesh_shape = (1, config.mesh.fsdp, config.mesh.tensor, config.mesh.expert)
```

Comment:
Can we check that `num_devices` equals to the product of all these numbers, and raise an error if there's a mismatch? It seems likely someone might forget to override `mesh.fsdp` when running on large clusters.

Reply:
Yup, good idea